### ISSUE TITLE
GPG key on keys.gnupg.net expired

### ISSUE THREAD

**Issue by berendo** [NVIDIA/nvidia-docker#1386]
I am aware of the repository instructions, as well as #1081, and for first-time installation of the repository GPG key, we use:
```
curl -s -L https://nvidia.github.io/nvidia-docker/gpgkey | \
  sudo apt-key add -
```
However, for ongoing updates of keys, a scripted methodology around `apt-key` that does not have baked-in URLs is preferred and has worked in the past. Alas, right now the repository GPG key that's refreshed from `keys.gnupg.net` expired on `2020-09-17`:
```
berend@berend-test-1:~$ LANG=C apt-key list --keyid-format long -v 2>/dev/null | fgrep expired
sub   rsa2048/6ED91CA3AC1160CD 2019-09-18 [S] [expired: 2020-09-17]

berend@berend-test-1:~$ sudo apt-key adv --recv-keys --keyserver keys.gnupg.net 6ED91CA3AC1160CD
Executing: /tmp/apt-key-gpghome.Tzj5iCgtO9/gpg.1.sh --recv-keys --keyserver keys.gnupg.net 6ED91CA3AC1160CD
gpg: key DDCAE044F796ECB0: "NVIDIA CORPORATION (Open Source Projects) <cudatools@nvidia.com>" 1 new signature
gpg: key DDCAE044F796ECB0: "NVIDIA CORPORATION (Open Source Projects) <cudatools@nvidia.com>" 1 new subkey
gpg: Total number processed: 1
gpg:            new subkeys: 1
gpg:         new signatures: 1

berend@berend-test-1:~$ LANG=C apt-key list --keyid-format long -v
/etc/apt/trusted.gpg
--------------------
...
pub   rsa4096/DDCAE044F796ECB0 2017-09-28 [SCE]
      Key fingerprint = C95B 321B 61E8 8C18 09C4  F759 DDCA E044 F796 ECB0
uid                 [ unknown] NVIDIA CORPORATION (Open Source Projects) <cudatools@nvidia.com>
sub   rsa2048/6ED91CA3AC1160CD 2019-09-18 [S] [expired: 2020-09-17]
sub   rsa2048/C45B1676A04EA552 2017-09-28 [S] [expired: 2019-09-20]

gpg: Note: signature key 6ED91CA3AC1160CD expired Thu Sep 17 13:16:16 2020 PDT
gpg: Note: signature key C45B1676A04EA552 expired Fri Sep 20 16:41:19 2019 PDT
...
```
The more verbose output from `apt-key list` pre and post update are:
```
berend@berend-test-1:~$ LANG=C apt-key list --keyid-format long
/etc/apt/trusted.gpg
--------------------
...
pub   rsa4096/DDCAE044F796ECB0 2017-09-28 [SCE]
      Key fingerprint = C95B 321B 61E8 8C18 09C4  F759 DDCA E044 F796 ECB0
uid                 [ unknown] NVIDIA CORPORATION (Open Source Projects) <cudatools@nvidia.com>
sub   rsa2048/6ED91CA3AC1160CD 2019-09-18 [S] [expired: 2020-09-17]
...
```
```
berend@berend-test-1:~$ LANG=C apt-key list --keyid-format long
/etc/apt/trusted.gpg
--------------------
...
pub   rsa4096/DDCAE044F796ECB0 2017-09-28 [SCE]
      Key fingerprint = C95B 321B 61E8 8C18 09C4  F759 DDCA E044 F796 ECB0
uid                 [ unknown] NVIDIA CORPORATION (Open Source Projects) <cudatools@nvidia.com>
sub   rsa2048/6ED91CA3AC1160CD 2019-09-18 [S] [expired: 2020-09-17]
sub   rsa2048/C45B1676A04EA552 2017-09-28 [S] [expired: 2019-09-20]
...
```
tl;dr is it intended that the repository GPG key published on `keys.gnupg.net` are kept up to date with respect to the one downloadable via https://nvidia.github.io/nvidia-docker/gpgkey? If not, should the key even be on `keys.gnupg.net`?

**Comment by berendo** [comment]
Any comments?

**Comment by guptaNswati** [comment]
Hi Berend

On my end, the key still seems to be active

```
pub   4096R/DDCAE044F796ECB0 2017-09-28
uid                          NVIDIA CORPORATION (Open Source Projects) <cudatools@nvidia.com>
sub   2048R/C45B1676A04EA552 2017-09-28 [expired: 2019-09-20]
sub   2048R/6ED91CA3AC1160CD 2019-09-18 [expires: 2021-06-16]
```

**Comment by berendo** [comment]
> On my end, the key still seems to be active

@guptaNswati are you talking about freshly-installed keys (via `curl -s -L https://nvidia.github.io/nvidia-docker/gpgkey`) or ones updated (via `apt-key adv --recv-keys --keyserver keys.gnupg.net`) once the initial install has been done? There's zero doubt freshly-installed keys fetched from the repository are current. This issue is about the keys that are published on `keys.gnupg.net` for ongoing updates.

**Comment by guptaNswati** [comment]
Okay yes, its expired. Will update it. Thanks.

**Comment by berendo** [comment]
> Okay yes, its expired. Will update it. Thanks.

No problem - thank you for looking into it.

**Comment by guptaNswati** [comment]
We have decided to remove the keys on GNU server. Thanks for bringing this up. Closing for now.
